### PR TITLE
jMeter performance testing for Settlement.

### PR DESCRIPTION
**Pull request by koekiebox**

### CODE REVIEW

File: packages/api-svc/src/routes.ts [hunk 1]
```diff
@@ -69,7 +70,7 @@ const MAX_ENTRIES_PER_PAGE = 100;
 export class ExpressRoutes {
 	private readonly _logger: ILogger;
 	private readonly _tokenHelper: ITokenHelper;
-	// private readonly _aggregate: SettlementsAggregate;
+	private readonly _aggregate: SettlementsAggregate;
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/routes.ts:73]: We should limit the usage of the domain code from the CommandHandlerSvc

File: packages/api-svc/src/routes.ts [hunk 2]
```diff
@@ -102,51 +105,50 @@ export class ExpressRoutes {
 		// Inject authentication - all requests require a valid token.
 		this._router.use(this._authenticationMiddleware.bind(this)); // All requests require authentication.
 		
-		// transfer inject
-		// this is for tests only, normal path is though events (event/command handler)
-		// this._router.post("/transfer", this.postHandleTransfer.bind(this));
+		// Transfer inject
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/routes.ts:108]: Pls remove if not needed

File: packages/api-svc/src/routes.ts [hunk 3]
```diff
@@ -188,17 +190,36 @@ export class ExpressRoutes {
 		return this._router;
 	}
 
+	// this is for tests only, normal path is though events (event/command handler)
+	private async postHandleTransfer(req: express.Request, res: express.Response): Promise<void> {
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/routes.ts:194]: Not needed anymore

File: packages/api-svc/src/service.ts [hunk 4]
```diff
@@ -171,7 +212,9 @@ export class Service {
 		}
 		globalLogger = this.logger = logger;
 
-		if(!tokenHelper){
+		if (!tokenHelper && bareboneStartup) {
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:215]: This code should only check if an implementation was provided, if not use the default. 
For production, the Service.Start() should be called with empty params.
For tests, the Service.Start() should inject the mocks via its params.
(same for others below)

File: packages/api-svc/src/service.ts [hunk 5]
```diff
@@ -316,30 +379,43 @@ export class Service {
...
+		if (!configCache) {
+			configCache = new SettlementConfigCacheRepoMock();
+		}
+		configCache.init();
+		this.configCache = configCache;
+
+		if (!batchCache) {
+			batchCache = new SettlementBatchCacheRepoMock();
+		}
+		batchCache.init();
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:391]: Missing an await

File: packages/api-svc/src/service.ts [hunk 6]
```diff
@@ -316,30 +379,43 @@ export class Service {
 		}
 		this.metrics = metrics;
 
-		/* Aggregate cannot be used outside the command-handler
+		if (!configCache) {
+			configCache = new SettlementConfigCacheRepoMock();
+		}
+		configCache.init();
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:385]: Missing an await (and add redis implementation)

File: packages/api-svc/src/service.ts [hunk 7]
```diff
@@ -382,16 +458,17 @@ export class Service {
 		if (this.batchRepo) await this.batchRepo.destroy();
 		if (this.batchTransferRepo) await this.batchTransferRepo.destroy();
 		if (this.matrixRepo) await this.matrixRepo.destroy();
-		if (this.participantAccountNotifier) await this.participantAccountNotifier.destroy();
 		if (this.logger instanceof KafkaLogger) await this.logger.destroy();
 	}
-}
 
+	static isEnvBarebone(): boolean {
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:464]: Probably don't need this anymore

File: packages/command-handler-svc/src/service.ts [hunk 8]
```diff
@@ -83,14 +83,25 @@ import {PrometheusMetrics} from "@mojaloop/platform-shared-lib-observability-cli
 import configClient from "./config";
 import {DEFAULT_SETTLEMENT_MODEL_ID, DEFAULT_SETTLEMENT_MODEL_NAME} from "@mojaloop/settlements-bc-public-types-lib";
 import {IConfigurationClient} from "@mojaloop/platform-configuration-bc-public-types-lib";
-import { ConfigurationClient,IConfigProvider } from "@mojaloop/platform-configuration-bc-client-lib";
+
 import crypto from "crypto";
+import {
+	AccountsBalancesAdapterVoid,
```
Comment (pedrosousabarreto) [inline packages/command-handler-svc/src/service.ts:89]: Mocks should only from tests, see Service.Start() comment on the API svc

File: packages/api-svc/src/service.ts [hunk 9]
```diff
@@ -123,41 +142,63 @@ const kafkaProducerOptions: MLKafkaJsonProducerOptions = {
 
 let globalLogger: ILogger;
 
+// tiger_beetle:
+const USE_TIGERBEETLE = process.env["USE_TIGERBEETLE"] || false;
+const TIGERBEETLE_CLUSTER_ID = process.env["TIGERBEETLE_CLUSTER_ID"] || 0;
+const TIGERBEETLE_CLUSTER_REPLICA_ADDRESSES = process.env["TIGERBEETLE_CLUSTER_REPLICA_ADDRESSES"] || "localhost:9001";
+
+const USE_MONGO_ADAPTER = process.env["USE_MONGO_ADAPTER"] || false;
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:150]: Not needed, should be injection via Service.Start( params)

File: packages/api-svc/src/service.ts [hunk 10]
```diff
@@ -123,41 +142,63 @@ const kafkaProducerOptions: MLKafkaJsonProducerOptions = {
...
-	static participantAccountNotifier: IParticipantAccountNotifier;
 	static batchTransferRepo: ISettlementBatchTransferRepo;
 	static matrixRepo: ISettlementMatrixRequestRepo;
 	static metrics: IMetrics;
 	static messageProducer: IMessageProducer;
 	static startupTimer: NodeJS.Timeout;
+	static abAdapter: IAccountsBalancesAdapter;
+	static configCache: ISettlementConfigCacheRepo;
+	static batchCache: ISettlementBatchCacheRepo;
+	static aggregate: SettlementsAggregate;
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:177]: The aggregate is not longer needed here, probably some aggregate only dependencies can also be removed (like the cache)

File: packages/command-handler-svc/src/service.ts [hunk 11]
```diff
@@ -367,23 +406,44 @@ export class Service {
 		}
 		this.metrics = metrics;
 
+		if (!configCache) {
+			configCache = new SettlementConfigCacheRepoMock();
+		}
+		configCache.init();
```
Comment (pedrosousabarreto) [inline packages/command-handler-svc/src/service.ts:412]: Missing the await. Could be redis implementation?

File: packages/command-handler-svc/src/service.ts [hunk 12]
```diff
@@ -367,23 +406,44 @@ export class Service {
...
+		this.configCache = configCache;
+
+		if (!batchCache) {
+			batchCache = new SettlementBatchCacheRepoMock();
+		}
+		batchCache.init();
```
Comment (pedrosousabarreto) [inline packages/command-handler-svc/src/service.ts:418]: Missing the await. Could be redis implementation?

File: packages/command-handler-svc/src/handler.ts [hunk 13]
```diff
@@ -208,15 +218,37 @@ export class SettlementsCommandHandler{
...
 				this._logger.error(err, `TransfersCommandHandler - processing command - ${message?.msgName}:${message?.msgKey}:${message?.msgId} - Error: ${(err as Error)?.message?.toString()}`);
-			}finally {
+			} finally {
 				resolve();
 			}
 		});
 	}
 
 	private async _getServiceSecContext():Promise<CallSecurityContext>{
+		if (this._barebone) {
```
Comment (pedrosousabarreto) [inline packages/command-handler-svc/src/handler.ts:230]: We don't need to check security context in the aggregate functions called by the command handler (as they are all system calls).
For now, we should only check for privileges in the API (create matrix, lock, etc..)

**Review by pedrosousabarreto — CHANGES_REQUESTED** [review]
This is looking great! Lot of work done here, thanks!

**Comment by koekiebox** [comment]
@pedrosousabarreto managed to work through the feedback. Will get the Redis work in by Monday morning for a final 👁️

File: packages/api-svc/src/service.ts (hunk 5 above)
Comment (koekiebox) [reply packages/api-svc/src/service.ts]: Fixed.

File: packages/command-handler-svc/src/service.ts (hunk 12 above)
Comment (koekiebox) [reply packages/command-handler-svc/src/service.ts]: Fixed. Redis has now been added (`SettlementBatchCacheRepoRedis`): 
`packages/infrastructure-lib/src/cache/settlement_batch_cache_repo_redis.ts`

**Review by pedrosousabarreto — APPROVED** [review]
Looks good to me. Thanks @koekiebox

**Pull request merged**